src/function/gds/shortest_paths.cpp

```diff
@@ -310,12 +426,23 @@ class ShortestPathsAlgorithm final : public GDSAlgorithm {
 
     void runShortestPathsFromSource(processor::ExecutionContext* executionContext,
         nodeID_t sourceNodeID) {
-        auto sourceState = ShortestPathsSourceState(sharedState->graph.get(), sourceNodeID);
-        auto pathLengthFrontiers =
-            std::make_shared<PathLengthsFrontiers>(sourceState.pathLengths.get());
-        ShortestPathsFrontierCompute spFrontierCompute(pathLengthFrontiers.get());
-        GDSUtils::runFrontiersUntilConvergence(executionContext, *pathLengthFrontiers,
-            sharedState->graph.get(), spFrontierCompute,
+        auto sourceState = ShortestPathsSourceState(sharedState->graph.get(), sourceNodeID,
+            outputType, executionContext->clientContext->getMemoryManager());
+
+        std::unique_ptr<Frontiers> frontiers;
+        std::unique_ptr<FrontierCompute> frontierCompute;
+        if (outputType == RecJoinOutputType::LENGTHS) {
+            frontiers = std::make_unique<PathLengthsFrontiers>(sourceState.pathLengths.get());
+            frontierCompute = std::make_unique<SPLengthsFrontierCompute>(
+                reinterpret_cast<PathLengthsFrontiers*>(frontiers.get()));
```

Comment:
Avoid using reinterpret_cast as long as u can. Search for ptrCast in the code base and mimic the design.

reinterpret_cast is not safe as it will blindly cast without perform any type/memory check. Use reinterpret_cast only when the base type is unknown, e.g. when u declare a unit8_t* buffer.

Reply:
I can't put ptrCast to these functions directly because frontiers is of class Frontiers* and that's the base abstract class. I need to cast to the parent class PathLengthsFrontiers*. But I found a bit more verbose way to achieve the same thing without reinterpret cast.